rc/lsp.kak

```diff
@@ -342,9 +342,39 @@ have_kakoune_feature_filtertext = ${kak_opt_lsp_have_kakoune_feature_filtertext}
 " | eval "${kak_opt_lsp_cmd} --request") > /dev/null 2>&1 < /dev/null & }
 }}
 
+declare-option -hidden str lsp_completion_selection ""
+
 define-command -hidden lsp-completion-dismissed -docstring "Called when the completion pager is closed" %{
+    set-option window lsp_completion_selection %val{hook_param}
+    try %{
+        eval -draft %{
+            select %opt{lsp_completion_selection}
+        }
+    } catch %{
+        # Completion was cancelled, nothing was inserted
+        # Clear these so the following lines do nothing.
+        # This is to avoid runing them in the above try block, and silencing their errors
+        remove-hooks window lsp-completion-dismissed
+        set-option window lsp_completions_selected_item -1
+    }
     lsp-completion-item-resolve-request false
+    trigger-user-hook "LspCompletionDismissed"
```

Comment:
I'm not sure if a user hook is the simplest solution (I'd need to look up why user hooks exist)
Can we instead use a window-scoped option? I feel like that'll end up being less complicated

Reply:
User hooks are made for this purpose - i.e. to implement custom hooks, without having to use a dummy option and a SetOption hook. However, since we already have options that are set here, it might be possible to simply detect WinSetOption for `lsp_completions_selected_item` for example.

Reply:
The original request for user hooks is in [this post](https://discuss.kakoune.com/t/ann-coqoune/1127) where they want to run user hooks when a plugin creates a new window.
This allows users of the plugin to customize window placement by defining the corresponding user hooks.

OTOH this patch doesn't intend for users to define a custom `LspCompletionDismissed` hook, so we might as well set a `declare-option -hidden str lsp_completion_on_dismiss` and evaluate it in our `InsertCompletionHide` hook.